### PR TITLE
tvOS compatibility

### PR DESCRIPTION
**Pull request by joonmanji**
# Summary

We have a monorepo that shares podfiles across mobile and tv apps. We'd like to make certain libraries tvOS compatible

Currently affects:
- Contacts
- Notifications
- Photo Library

## Test Plan

Projects should still compile with iOS code unaffected and will now compile for tvOS.

### What's required for testing (prerequisites)?

no prereqs

### What are the steps to reproduce (after prerequisites)?

does it compile?

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅    |
| Android |    ✅     |

## Checklist

- [X] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

**Review by mikehardy — COMMENTED** [review]
Not sure if you've ever hacked in tvOS support but this looks about how it's gone in other repos I've worked in @zoontek :-) - mostly just ifdef'ing it out as short circuits.

Might be nice to have a generic "make me an error with this code+message" method somewhere to reduce duplication but :shrug: 

Might be nice to just chuck all ideas of formatting at the clang-format program as a separate matter https://github.com/invertase/react-native-firebase/blob/514e6bd51e7624a6403dda706f4e5b65cee63422/package.json#L12-L13 (clang-format is an npm package that delivers the binary in usable format nicely)

all minor though, the general ifdef'ing strategy is all I've seen done normally so works for me FWIW

File: ios/PhotoLibrary/RNPermissionHandlerPhotoLibrary.m [hunk 1]
```diff
@@ -16,6 +16,9 @@ + (NSString * _Nonnull)handlerUniqueId {
 
 - (void)checkWithResolver:(void (^ _Nonnull)(RNPermissionStatus))resolve
                  rejecter:(void (__unused ^ _Nonnull)(NSError * _Nonnull))reject {
+#if TARGET_OS_TV
+    return reject(@"cannot_open_limited_picker", @"Photo Library not available on tvOS", nil);
```
Comment (zoontek) [inline ios/PhotoLibrary/RNPermissionHandlerPhotoLibrary.m:20]: Wrong message code: `cannot_check_permission`

File: ios/Contacts/RNPermissionHandlerContacts.m [hunk 2]
```diff
@@ -14,6 +16,13 @@ + (NSString * _Nonnull)handlerUniqueId {
 
 - (void)checkWithResolver:(void (^ _Nonnull)(RNPermissionStatus))resolve
                  rejecter:(void (__unused ^ _Nonnull)(NSError * _Nonnull))reject {
+#if TARGET_OS_TV
```
Comment (zoontek) [inline ios/Contacts/RNPermissionHandlerContacts.m:19]: You mix up `NSError` based and string based rejection. Maybe you can keep it to the second?

```objc
#if TARGET_OS_TV
    NSError * error = [[NSError alloc]
                       initWithDomain:@"RNPermissions"
                       code:1
                       userInfo:@{NSLocalizedDescriptionKey: @"Contacts not available on tvOS"}];
    return reject(error);
#elif

// ⬇️

#if TARGET_OS_TV
  return reject(@"cannot_check_permission", @"Contacts not available on tvOS", nil);
#elif
```

**Review by zoontek — CHANGES_REQUESTED** [review]

**Comment by zoontek** [comment]
I'm closing this since there were no follow up.

**Pull request closed without merging**